gaphor/storage/storage.py

```diff
@@ -153,8 +156,12 @@ def update_status_queue() -> Iterable[float]:
         if progress % 30 == 0:
             yield (progress * 100) / size
 
-    # First create elements and canvas items in the factory
+    # First ensure that all sysml diagrams have a frame
+    if version_lower_than(gaphor_version, (2, 22, 0)):
+        ensure_sysml_diagrams_have_a_frame(elements)
```

Comment:
I am not sure whether it is taking place at the correct step in model loading. The way I implemented it; it has to happen before we start loading elements and items.

Another option is to do it afterwards, but replacing an item is easier to be done by simply changing the type string rather than to rewire everything to a new item.

Reply:
I would do it as early as possible.

What would happen if we leave out this upgrade? Is it required to do this upgrade or existing models just remain as is?

Reply:
Since it is not possible to add diagram frame item explicitly later, existing SysML diagrams need to get the item at minimal. I think I'll go with just adding frame item on SysML diagrams to surround all the elements there and leave the other refactoring for the users. Seems to be the safest option. Then you can choose when to start utilizing the frame for parameters/ports. I suppose we could leave a guide or add blog entry about this as well on how to utilize SysML frame.

Another alternative is to perform no upgrade, but in order to get diagram frame, you need to create brand new diagram.